feat: Long-running operation improvements for `mongodbatlas_cluster_outage_simulation` resource (#3541)
mongodb/terraform-provider-mongodbatlas: .changelog/3541.txt

```release-note:enhancement

> resource/mongodbatlas_cluster_outage_simulation: Adds `delete_on_create_timeout` attribute to indicate whether to delete the resource if its creation times out

Reviewer: ```suggestion
resource/mongodbatlas_cluster_outage_simulation: Indicates whether to delete the created resource if a timeout is reached when waiting for completion.
```
Let's rephrase as per @AgustinBettati's suggestion in this [PR](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3525#discussion_r2242502637).

Author: Great, sound good! will do the changes as part of CLOUDP-335200 so that I do the change once everywhere
